### PR TITLE
feat: StackedDiscreteBar charts tooltips and they show values inside bars

### PR DESCRIPTION
**Pull request by MarcelGerber**
Notion: [Show values and tolerance intervals](https://www.notion.so/Show-values-and-tolerance-intervals-cbfc40b4f0c24c7282be6301d51ab247)

This adds two important features to Stacked Discrete Bar charts:
* Bars are labeled with their values, if the bar is big enough
* Upon hover over a bar or a country name, a tooltip shows

This is also live on _tufte_. Try out these two:
* https://tufte-owid.netlify.app/grapher/per-capita-energy-stacked
* https://tufte-owid.netlify.app/covid-vaccinations#what-share-of-the-population-has-been-partly-or-fully-vaccinated-against-covid-19

It's important to note that this is the first tooltip that uses Tippy, which makes things so much easier and also looks a little nicer. This also means that tooltips can go outside the chart frame.

## Screenshots
image
image

### CODE REVIEW

**Comment by MarcelGerber** [comment]
@danielgavrilov I tweaked this a bit here and there, but I should be all done now and this is now ready for review 🚀

File: grapher/stackedCharts/StackedDiscreteBarChart.tsx [hunk 1]
```diff
@@ -362,6 +379,209 @@ export class StackedDiscreteBarChart
...
+        const barX = axis.place(this.x0 + point.valueOffset)
+        const barWidth = axis.place(point.value) - axis.place(this.x0)
+
+        // Compute how many decimal places we should show.
+        // Basically, this makes us show 2 significant digits, or no decimal places if the number
+        // is big enough already.
+        const dp = Math.ceil(-Math.log10(point.value) + 1)
+        const barLabel = formatColumn.formatValueShort(point.value, {
+            numDecimalPlaces: isFinite(dp) && dp >= 0 ? dp : 0,
+        })
```
Comment (danielgavrilov) [inline grapher/stackedCharts/StackedDiscreteBarChart.tsx:397]: Looks very similar to https://github.com/owid/owid-grapher/blob/9bc2033bbdc0e834d0d8589a080232d602e6f84e/grapher/axis/Axis.ts#L243-L245

Could some util be extracted?

File: grapher/stackedCharts/StackedDiscreteBarChart.tsx [hunk 2]
```diff
@@ -362,6 +379,209 @@ export class StackedDiscreteBarChart
...
+        const labelBounds = Bounds.forText(barLabel, {
+            fontSize: 12.6,
```
Comment (danielgavrilov) [inline grapher/stackedCharts/StackedDiscreteBarChart.tsx:399]: Why is the `fontSize` hardcoded? Shouldn't it depend on `baseFontSize`?

File: grapher/stackedCharts/StackedDiscreteBarChart.tsx [hunk 3]
```diff
@@ -362,6 +379,209 @@ export class StackedDiscreteBarChart
...
+                        <td colSpan={4} style={{ color: "#111" }}>
+                            <strong>{props.label}</strong>
+                        </td>
+                    </tr>
+                    {props.bars.map((bar) => {
+                        const { highlightedSeriesName } = props
+                        const squareColor = bar.color
+                        const isHighlighted =
+                            bar.seriesName === highlightedSeriesName
+                        const isFaded =
```
Comment (danielgavrilov) [inline grapher/stackedCharts/StackedDiscreteBarChart.tsx:470]: nit: `isFaded` and `isFaint` have the same intended meaning?

**Review by danielgavrilov — APPROVED** [review]
Looks awesome! ✨ Left some nitpickings, but the functionality/interaction works really well!

In case @vanwoerden hasn't seen this in its final form, I'd just wait for his approval/comments (I asked him in the slack PR message)

File: grapher/stackedCharts/StackedDiscreteBarChart.tsx (hunk 1 above)
Comment (MarcelGerber) [reply grapher/stackedCharts/StackedDiscreteBarChart.tsx:397]: In theory, yes, and this is also where I took this from.
However, I wouldn't know what to name that? `detectFirstSignificantDigit` maybe, but then you would also need to clarify what a return value of 2 means?

Somewhat related, I was actually thinking of the [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat) API while working on this. They combine the options `minimumFractionDigits`, `minimumSignificantDigits`, `maximumFractionDigits`, and `maximumSignificantDigits` to achieve a very similar effect that we do in a different way. Maybe our number formatting should also work this way, or even use the `Intl.NumberFormat` API internally (using only the English preset)?

File: grapher/stackedCharts/StackedDiscreteBarChart.tsx (hunk 2 above)
Comment (MarcelGerber) [reply grapher/stackedCharts/StackedDiscreteBarChart.tsx:399]: Good catch!

File: grapher/stackedCharts/StackedDiscreteBarChart.tsx (hunk 3 above)
Comment (MarcelGerber) [reply grapher/stackedCharts/StackedDiscreteBarChart.tsx]: Yeah, and they even sound so similar!

**Comment by MarcelGerber** [comment]
Thank you for your review @danielgavrilov, I've addressed the nits you had. Let's get this out there!

In Notion, [I've also listed some extensions](https://www.notion.so/owid/Show-values-and-tolerance-intervals-cbfc40b4f0c24c7282be6301d51ab247#70364351122e4efa96fdcf17cd3f46dd) that we could add in the future.

**Pull request merged**